src/shell/layout/tiling/mod.rs

```diff
@@ -1303,9 +1303,14 @@ impl TilingLayout {
             .unwrap()
             .clone()
             .ok_or(NodeIdError::NodeIdNoLongerValid)?;
+
+        let tree = &self.queue.trees.back().unwrap().0;
+        let tree_node = tree
+            .get(&node_id)
+            .map_err(|_| NodeIdError::NodeIdNoLongerValid)?;
+
         let state = {
-            let tree = &self.queue.trees.back().unwrap().0;
-            tree.get(&node_id).unwrap().parent().and_then(|parent_id| {
+            tree_node.parent().and_then(|parent_id| {
```

Comment:
This is not a correct fix, the `unwrap` is very much there on purpose to catch these logic errors, we should never try to unmap an element, that is not part of the layout.

Do you have a backtrace of this crash? We would want to figure out instead, where the `unmap` call comes from and why it ends up being called wrongly.